File: src/main/java/fds/java/consistent/HashRing.java
package fds.java.consistent;

/**
 * Stub for a Consistent hash ring
 */
public class HashRing {

    // TODO create data structure for the hash ring

    public HashRing( int size ) {
        // TODO set up the ring of the given size
    }


    public RingNode addNode(RingNode node) {
        // TODO add a ring node to this ring
        return null;
    }

    public RingNode removeNode(RingNode node) {
        // TODO remove a ring node from this ring
        return null;
    }

    public void put(String  key, Object  value) {
        // TODO find the appropriate node and add the value under the key
    }

    public Object get( String key) {
        // TODO find the appropriate node and get the value given the key
        return null;
    }

    public int size() {
        // TODO Total the number of
        return 0;
    }

    private RingNode walkRingToFindNextNode( int startingLocation) {
        // TODO walk round the ring until you hit a node
        return null;
    }

}

File: src/main/scala/fds/scala/consistent/HashRing.scala
package fds.scala.consistent

/**
  * Stub for a Consistent hash ring
  */
class HashRing(slots: Int) {

  // TODO create data structure for the hash ring

  def addNode(node: RingNode):  RingNode = {
    // TODO add a ring node to this ring
    null
  }

  def removeNode(node: RingNode): RingNode = {
    // TODO remove a ring node from this ring
    null
  }

  def put(key: String, value: Any): Unit = {
    // TODO find the appropriate node and add the value under the key
  }

  def get(key: String): Any = {
    // TODO find the appropriate node and get the value given the key
    null
  }

  def size: Int = {
    // TODO Total of all nodes' sizes
    0
  }

  def walkRingToFindNextNode(startingLocation: Int): RingNode = {
    // TODO walk round the ring until you hit a node
    null
  }
}

object HashRing {
  def apply(slots: Int): HashRing = new HashRing(slots)
}

